### ISSUE TITLE
[3.5.0] Release a new version 

### ISSUE THREAD

**Issue by rperezb** [owncloud/ios#663]
#TASKS:
- [X] [GIT] Create branch release3.5.0 (freeze the code) @mRoRo
- [X] [GIT] Create the branch release_2.0.0_library @mRoRo
- [X] [DIS] Update changelog library
- [X] [DIS] Update changelog ios app
- [X] [DEV] Update version number 3.5.0 @nasli 
- [X] [DEV] Update third party file with new library version @nasli
- [X] [DEV] Update readme library file
- [X] [DEV] Make the "3 dots" in Files view a branded element so that it could be black or white https://github.com/owncloud/ios/issues/663#issuecomment-222494341 [check if is not already brandeable]-->[Already brandeable, nothing to do]
- [X] [DEV] Refactor generate image name on instant uploads in order to use the same that the new one created for the new uploads inside the app [FIXED via  https://github.com/owncloud/ios/pull/702#discussion_r67012613] @Jon-Schneider 
- [X] [QA] Design Test plan @jesmrec
- [X] [QA] Regression Test plan @mcastroSG @jesmrec 
- [X] [DIS] Update screenshots. At least, iPad Pro ones @nasli
- [X] [DIS] Release to appstore @nasli
- [X] [GIT] Merge branch release3.5.0 @nasli
- [X] [GIT] Merge branch release_2.0.0_library @nasli
- [X] [GIT] Create tag and sign it @nasli
- [X] [DEV] Update error text when the mail is not configured  and you select to send feedback, currently: Your device doesn't support the composer sheet. -> expected: "Please, configure your mail account" [FIXED] @nasli
- [X] [DEV] Improve search users performance in share view https://github.com/owncloud/ios/issues/638#issuecomment-233329575 https://github.com/owncloud/ios/issues/663#issuecomment-235302668 @nasli 
- [X] [DEV] Add Chinese and other languages [https://github.com/owncloud/ios/issues/735] @nasli
- [X] [DEV] Add Chinese(china)" zh_CN, "Chinese (Taiwan)" zh_TW , "Albanian" sq  and "Slovenian" sl in Jenkins [@DeepDiver1975]
- [X] [DEV] Bring changes from master after changes transifex done and add Chinese new files into project, fixed Thai language @nasli
- [X] [GIT] Merge faq file with new issue https://github.com/owncloud/ios/pull/658 
- [X] [DIS] Update App Store information @nasli [updated support forum]

BRANDING CHANGES
- [X] [OB] Modify branded compilation to include and change parameters related to status bar https://github.com/owncloud/ios/issues/663#issuecomment-223055754
- [X] [OB] 3 dots icons menu [maybe not needed and already brandeable]-->[Already brandeable, nothing to do]
- [X] [OB] update HelpImage for FileDownloadedIcon (pending merge ownbrander PR after release)
- [X] [DEV] replace localized strings with the correct $appname, related PR #715 , https://github.com/owncloud/ios/issues/663#issuecomment-232424599 @mRoRo [FIXED]
- [X] "$appname" text is shown in the first help guide view. Same for "error_login_doc_provider" string https://github.com/owncloud/ios/issues/663#issuecomment-232401645 @mRoRo [FIXED]

REGRESSION BUGS FOUND:
- [X] Favorite folders are not shown as favorite in the Share Sheet https://github.com/owncloud/ios/issues/663#issuecomment-223923105 [WON'T FIX, works as expected, in the extensions this feature is not implemented since not is the scope of these view]
- [X] Uploads list cleared https://github.com/owncloud/ios/issues/663#issuecomment-230419398 @mRoRo [FIXED] 
- [X] Download files using document provider https://github.com/owncloud/ios/issues/663#issuecomment-231289776 @mRoRo [FIXED]
- [X] Video upload from the asset upload picker is broken  [FIXED via https://github.com/owncloud/ios/issues/699] @Jon-Schneider 
- [X] Uploads error after session expired https://github.com/owncloud/ios/issues/663#issuecomment-232659486 @nasli [FIXED]
- [X] "Background Instant Uploads" switch in settings tab is not enabled the first time it's enabled and the user grants permission. https://github.com/owncloud/ios/issues/663#issuecomment-233876760 [FIXED https://github.com/owncloud/ios/pull/740] @Jon-Schneider 
- [X] Instant uploads put photos on My Photo Stream Album and provokes duplicate uploads in case of two devices using the same account. https://github.com/owncloud/ios/issues/663#issuecomment-233590512 [FIXED via https://github.com/owncloud/ios/pull/718/commits/f50254008246f58f4e576c93e4d9fa90900f7a47] @Jon-Schneider 
- [X] Race condition for upload available offline files https://github.com/owncloud/ios/issues/663#issuecomment-233656327 [NOT REPRODUCEABLE]
- [X] Enable instant uploads without permissions on Photos hangs the app https://github.com/owncloud/ios/issues/663#issuecomment-234227555 [FIXED https://github.com/owncloud/ios/pull/742] @Jon-Schneider 
- [X] Crash preview files in ipad with share in server disabled  https://github.com/owncloud/ios/issues/663#issuecomment-234248716 @nasli [FIXED]
- [X] SAML sessions in iPad removes local copy of av. offline files https://github.com/owncloud/ios/issues/663#issuecomment-235304981 @nasli [FIXED]
- [X] Instant uploads of live photos duplicate photos https://github.com/owncloud/ios/issues/746 @Jon-Schneider  [FIXED] https://github.com/owncloud/ios/pull/750
- [X] SAML downloads inside folder marked as finished after a expiration https://github.com/owncloud/ios/issues/663#issuecomment-236111818 @nasli [FIXED]
- [X] SAML expiration asks always for accept certificate https://github.com/owncloud/ios/issues/663#issuecomment-236112261 [FIXED] @nasli
- [X] SAML downloads individuals with priority marked as downloading after a expiration  @nasli [FIXED] Canceled after session renewed
- [X] SAML instant uploads duplicate files and cause conflicts after expiration https://github.com/owncloud/ios/issues/663#issuecomment-238523357 [FIXED] @nasli
- [X] Login view freezes after SAML session expiration https://github.com/owncloud/ios/issues/663#issuecomment-238801664 [NO REPRODUCEABLE] 
- [X] SSO does not show authentication view https://github.com/owncloud/ios/issues/663#issuecomment-238832049 @nasli [FIXED] Added control to redirects and check if new certificates is needed in sso view
- [x] http url does not stablish connection https://github.com/owncloud/ios/issues/663#issuecomment-240712849 [FIXED] @nasli

NO REGRESSION
- [X] Upload big file in background https://github.com/owncloud/ios/issues/663#issuecomment-231995554 @nasli [FIXED]
- [X] Resumed uploads do not finish https://github.com/owncloud/ios/issues/663#issuecomment-234949111 @nasli [SERVER ISSUE, the file and folder remain locked https://github.com/owncloud/core/issues/17254#issuecomment-235336676]
- [X] SAML Session expires when the account is switched https://github.com/owncloud/ios/issues/663#issuecomment-235507410 -> https://github.com/owncloud/ios/issues/747
- [X] Av. offline folders do not start to download if they are set after SAML session expires https://github.com/owncloud/ios/issues/663#issuecomment-237787368 [WON'T FIX] @nasli Relaunched after a refresh of the file list

PR:https://github.com/owncloud/ios/pull/733

**Comment by mRoRo** [comment]
![screen shot 2016-05-30 at 15 28 28](https://cloud.githubusercontent.com/assets/15103194/15650896/28714e7e-267c-11e6-9594-8aac5e20341f.png)

Add the "3 dots" (more) black image as branding option

**Comment by rperezb** [comment]
Good catch

**Comment by mRoRo** [comment]
**k_is_text_status_bar_white:** This constant has been modified. Its default value is YES now and it'll manage all status bars of the app's views (not extensions) but the logging view. **This constant was used previously to manage the status bar of the loggin view.**

**k_is_text_login_status_bar_white:** This constant is new and manages only the loggin status bar color. It's default vaule is NO. It makes what k_is_text_status_bar_white did before.

**Comment by rperezb** [comment]
Thanks @mRoRo yes, 
- k_is_text_status_bar_white is under control
- k_is_text_login_status_bar_white we will add it once the new version is released

**Comment by mRoRo** [comment]
Favorite folders are not shown as favorite in the Share Sheet.:

This behaviour is present in "ownCloud" and "Copy to ownCloud".

**Comment by jesmrec** [comment]
Bug: Uploads list is cleared if the app is killed

**Comment by jesmrec** [comment]
Bug: a non-downloaded file is not dowloaded if it is opened using the document provider.

**Comment by mcastroSG** [comment]
### Bug

### Steps

1. Try to upload a big file
2. Open another apps or let the mobile turn of the screen due to inactivity

### Actual behaviour

Upload fails and it´s restarted anytime you open ownCloud app.

### Expected behaviour

A proper background upload should be done.

Device: iPhone 6
IOS Version: 8.1.3

**Comment by mRoRo** [comment]
### Bug

### Steps

1. Remove previous version of ownCloud app.
2. Install the app showing the help (k_show_main_help_guide = YES)

### Actual behaviour

In the first view, it's show the text "$appname".

### Expected behaviour

There should be shown the app name (ownCloud for example).

Device: iPhone 5
iOS Version: 9.0

**Comment by nasli** [comment]
Replace $appname on this strings:
"error_login_doc_provider" 
"files_will_be_upload_next_time"
"title_help_slide_0"

**Comment by mcastroSG** [comment]
### Bug

### Steps

1. Upload several files and in the middle of the process expire Shib session


### Actual behaviour

App crash, and after that uploads fail due to a wrong conflict.

### Expected behaviour

Uploads should be completed and any upload that begins after shib expiration should not start until we renove credentials

Device: iPad Air
IOS Version: 8.1.2

**Comment by mcastroSG** [comment]
### Bug

### Steps

1. Set instant uploads on ownCloud app
2. Close app
3. Take some photos

### Actual behaviour

App upload any photo that we take but some conflict errors appears on uploads view due to seems to be uploading the same photo several times.

### Expected behaviour

ownCloud app should upload any photo we take without errors

Device: iPad Mini
IOS Version: 9.1

**Comment by mcastroSG** [comment]
@Jon-Schneider @nasli  executing some extra test I´ve found a new bug.

### Bug

**Executed Steps**

1.- Connect device1 to account1 on server1
2.- Connect device2 to account1 on server1
3.- Enable instant uploads on device1
4.- Enable instant uploads on device2
5.- Take a photo on device1

**Expected Result**

A new photo should be uploaded to account1 on server1 and after sync that photo should appear on InstantUploads folder on device1 and device2

** Actual behaviour **

Two photos are uploaded, one from device1 and other from device2. Both are the same photo but not same size. After talk with @nasli it seems that instant uploads does not upload photos from My Stream album in previous versions of owncloud app.

**Comment by mcastroSG** [comment]
### Bug

### Steps

1. Create a new txt file on app
2. Set this txt file as available offline
3. Edit the txt file on the app
4. Pull down to refresh the files view before upload process ends

### Actual behaviour

Upload fails due to conflict error, just sometimes, it´s hard to reproduce this bug.

### Expected behaviour

ownCloud app should upload txt file without problem.

Device: iPad Mini
IOS Version: 9.1

**Comment by Jon-Schneider** [comment]
@mcastroSG Check out the "Instant-Upload-using-modification-date" branch on my fork of ownCloud. This should fix the stream upload issue. (https://github.com/Jon-Schneider/ios/tree/Instant-Upload-using-modification-date). 

Let me know if the bug you noticed is still there. I haven't seen it - it may have been related to stream uploads and this patches it.

**Comment by mcastroSG** [comment]
@Jon-Schneider tested, now it works in a proper way. Thanks 😄

**Comment by mcastroSG** [comment]
### Bug

### Steps

1. Enable Instant Upload Backgrounds for first time
2. Answer yes to the new message asking for allow permissions

### Actual behaviour

In case of answer yes the option remains as inactive.

### Expected behaviour

In case of answer yes the option should appear active.

Device: iPad Mini
IOS Version: 9.1

**Comment by mcastroSG** [comment]
### Bug

### Steps

1. Enable Instant uploads on ownCloud app
2. Go to phone settings and remove access to Photos for ownCloud app
3. Go to Settings tab on ownCloud app and a new warning will appear informing about ownCloud does not have access to photos
4. Try to enable instant uploads 

### Actual behaviour

App seems hanged for 15 seconds till a new warning appears.

### Expected behaviour

A warning message informing why it´s not possible to set instant uploads should appear without waiting.

Device: iPad Mini
IOS Version: 9.1

**Comment by mcastroSG** [comment]
### Bug

### Steps

1. Disable Allow apps to use the Share API option as admin on server
2. Wait till share icon does not appear on preview view on app
3. Tap on an editable file as a txt

### Actual behaviour

App crashes.

### Expected behaviour

App do not crash, download the file and shows preview

Device: iPad Mini
IOS Version: 9.1

**Comment by jesmrec** [comment]
### Bug

### Steps

1. Upload a video or a huge file
2. One the upload has started, switch the device connection off
3. Wait a minute, and enable the device connection

### Actual behaviour

The upload is resumed but it does not end, remaining as "Waiting for server connection"

### Expected behaviour

The upload is resumed and ends correctly

Device: iPadAir v9, iPadMini v9.1
IOS Version: 9.1

cc @nasli

**Comment by jesmrec** [comment]
`Improve search users performance in share view`

A lag of one second is enough

cc @nasli

**Comment by jesmrec** [comment]
### Bug

### Steps

1. In a SAML server, set several files as available offline
2. Let the session expire
3. Login again

### Actual behaviour

The files are neither downloaded nor available offline

### Expected behaviour

The files remain as available offline

Device: iPadAir v8.2, iPadPro v9.1. Tested also with iPhone6 and works correctly.

cc @nasli

**Comment by nasli** [comment]
Updated delay @jesmrec https://github.com/owncloud/ios/issues/663#issuecomment-235302668

**Comment by nasli** [comment]
@jesmrec Updated video file name to keep same structure that photo files, using the apple id at the end

**Comment by jesmrec** [comment]
### Bug

### Steps

1. In a SAML server, sign in two accounts 
2. Open one of them
3. Switch to the another one

### Actual behaviour

The first operation makes the session expire altough the time is not expired. 

### Expected behaviour

Session does not expire until the lifetime is expired

Device: iPhone6 v9.3.

cc @nasli

**Comment by nasli** [comment]
https://github.com/owncloud/ios/issues/663#issuecomment-235507410 is a regression @jesmrec ?

**Comment by jesmrec** [comment]
@nasli no, moved to #747

**Comment by jesmrec** [comment]
### Bug

### Steps

1. Login in SAML server
2. Download a folder with videos or another content 
3. Let expire the session 
4. Login again

### Actual behaviour

The files are marked as downloaded, but they are not.

### Expected behaviour

The files keep on downloading. When they finish, they are marked as downloaded.

Tested with iPadAir v9, iPhone6 v9
Server v9.1

**Comment by jesmrec** [comment]
### Bug

### Steps

1. Login in SAML server with no trusted certificate
2. Accept the certificate 
3. Let expire the session 
4. Login again

### Actual behaviour

The certificate is asked after each session expiration

### Expected behaviour

Only the first time is asked

Tested with iPadAir v9, iPhone6 v9
Server v9.1

**Comment by jesmrec** [comment]
### Bug

### Steps

1. Login in SAML server
2. Wait until the session expires and set a folder as available offline
3. Pull down to expire the session
4. Login again

### Actual behaviour

The files into the folder are marked as downloading but they are not being downloaded unless you force it by tapping on one of them

### Expected behaviour

The files into the folder start to download after inputing the credential

Tested with iPadAir v8.4, iPhone6 v9.3.2
Server v9.1

**Comment by jesmrec** [comment]
### Bug

### Steps

1. Login in SAML server
2. Enable instant uploads and take some pics
3. Wait until session expires and login again

### Actual behaviour

Some conflicts appear in the uploads view, related to the instant uploads previously taken

### Expected behaviour

No conflicts related with the successful uploads 

Tested with iPadAir v9.1, iPhone v8
Server v9.1

cc @nasli @Jon-Schneider

**Comment by jesmrec** [comment]
### Bug

After a two or three SAML session expirations, the login view freezes and it is needed to kill the app and open it again.

The number of times to notice this behaviour is not always the same.  

Tested with iPadAir v8.4, iPhone v8 and v9.3
Server v9.1

**Comment by jesmrec** [comment]
### Bug

SAML server with SSO does not show the authentication webview. 

Tested with iPadAir v9.1, iPhone v8
Server v9.1

**Comment by jesmrec** [comment]
Most of the problems are fixed, but this one:

https://github.com/owncloud/ios/issues/663#issuecomment-235304981

in happening again 

cc @nasli

**Comment by jesmrec** [comment]
### Bug

###Steps:

1. In login view, set a url without prefix and http port. For example:

my.server.com:88976

###Actual behaviour:

"Testing connection" remains forever

###Expected behaviour

Connection is stablished

It works if the http prefix is set (for example, http://my.server.com:88976). For https servers works correctly with and without prefix

**Comment by davivel** [comment]
Last regression fixed. Ready to go.
